File: src/main/java/com/condominium/management/resident/entity/Resident.java
package com.condominium.management.resident.entity;

import com.condominium.management.unit.entity.Unit;
import com.condominium.management.user.entity.User;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "residents")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Resident {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String cpf;

    private String phone;

    private LocalDate birthDate;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false, unique = true)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "unit_id", nullable = false)
    private Unit unit;

    @Column(nullable = false)
    private LocalDateTime createdAt;
}

File: src/types/resident.ts
export interface Resident {
    id: number;
    userId: number;
    name: string;
    email: string;
    cpf: string;
    phone: string | null;
    birthDate: string | null;
    unitId: number;
    unitNumber: string;
}

export interface ResidentRequest {
    cpf: string;
    phone: string;
    birthDate: string;
    userId: number;
    unitId: number;
}